common/src/main/java/com/wynntils/wc/custom/item/IntelligenceSkillPointsItemStack.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.custom.item;
+
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import com.wynntils.wc.custom.item.properties.SkillPointProperty;
+import com.wynntils.wc.model.Character;
+import com.wynntils.wc.objects.SpellType;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+
+public class IntelligenceSkillPointsItemStack extends WynnItemStack {
+    private List<Component> tooltip;
+
+    public IntelligenceSkillPointsItemStack(ItemStack stack) {
+        super(stack);
+    }
+
+    @Override
+    public void init() {
+        List<Component> newTooltip = new ArrayList<>(getOriginalTooltip());
+        newTooltip.addAll(getTooltipExtension());
+        tooltip = newTooltip;
+    }
+
+    @Override
+    public List<Component> getTooltipLines(Player player, TooltipFlag isAdvanced) {
+        if (tooltip == null) {
+            return getOriginalTooltip();
+        }
+
+        return tooltip;
+    }
+
+    private List<Component> getTooltipExtension() {
+        List<Component> tooltipExtension = new ArrayList<>();
+        tooltipExtension.add(new TextComponent(""));
+        tooltipExtension.addAll(getManaTables());
+        return tooltipExtension;
+    }
+
+    private List<Component> getManaTables() {
+        Character.CharacterInfo characterInfo = WynnUtils.getCharacter().getCharacterInfo();
+        SkillPointProperty property = getProperty(ItemProperty.SKILL_POINT);
+        int intelligencePoints = property.getSkillPoints();
+
+        int closestUpgradeLevel = Integer.MAX_VALUE;
+        int level = characterInfo.getLevel();
+
+        LinkedList<Component> newLore = new LinkedList<>();
+
+        for (int j = 0; j < 4; j++) {
+            SpellType spell = SpellType.forClass(characterInfo.getClassType(), j + 1);
+
+            if (spell.getUnlockLevel(1) <= level) {
```

Comment:
is that meant to be 1?

Reply:
Yes, that means that the spell has not even been unlocked yet.

Reply:
Tbh I have no idea what that 1 means there but if you say so.